File: kotlin/1013-fibonacci-number-Easy/Solution.kt
class Solution {
    fun fib(n: Int): Int {
        return fibonacciSequence().elementAt(n)
    }
    
    fun fibonacciSequence() = sequence {
        var terms = Pair(0, 1)
        while (true) {
            yield(terms.first)
            terms = Pair(terms.second, terms.first + terms.second)
        }
    }
}

File: scala/1013-fibonacci-number-Easy/Solution.scala
object Solution {
    def fib(N: Int): Int = {
        val f = new Array[Int](Math.max(N + 1, 2))
        f(0) = 0
        f(1) = 1
        for (i <- 2 to N) f(i) = f(i - 1) + f(i - 2)
        f(N)
    }
}